### PR TITLE
Update various readme files to markdown

### PR DESCRIPTION
**Pull request by tamiko**
This commit renames various readme files to <code>README.md</code> and updates them to markdown. With this, the readme files are displayed on github when the respective subdirectory is chosen.

### CODE REVIEW

File: bundled/README.md [hunk 1]
```diff
@@ -1,49 +1,38 @@
...
-licenses below.
+licenses below.**
 
 Below is a detailed list of the content of each subdirectory, and the
 licenses that apply.
 
 
-./boost-1.49.0
-============
+./boost-1.56.0
```
Comment (tjhei) [inline bundled/README.md]: in another file you made directories monospace using backticks. Maybe here too?

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Further information:
+--------------------
+
+The testsuite documentation is located at ../doc/developers/testsuite.html.
```
Comment (tjhei) [inline tests/README.md]: backticks?

File: bundled/README.md (hunk 1 above)
Comment (bangerth) [inline bundled/README.md]: Also, best not to list individual version numbers given that they change over time. For example, below you still list 1.49.

**Comment by tjhei** [comment]
looks good otherwise

**Comment by tamiko** [comment]
Next iteration. I've omitted backticks in headings entirely and tried to unify the layout.

**Comment by tjhei** [comment]
Looks good to me. @bangerth any further comments?

File: bundled/README.md [hunk 3]
```diff
@@ -1,64 +1,54 @@
 This folder contains third party projects bundled with deal.II
+==============================================================
 
-PLEASE NOTE THAT THESE PROJECTS ARE COPYRIGHTED BY OTHERS THAN THE deal.II
+**PLEASE NOTE THAT THESE PROJECTS ARE COPYRIGHTED BY OTHERS THAN THE deal.II
 AUTHORS, but are included by permission. For details, consult the stated
-licenses below.
+licenses below.**
```
Comment (bangerth) [inline bundled/README.md]: This paragraph has a slightly ODD mixture OF lower AND upper CASE spelling. I know why you do it, it just looks odd. I guess it's not that big a deal, though.

File: bundled/README.md [hunk 4]
```diff
@@ -1,64 +1,54 @@
...
-http://threadingbuildingblocks.org/.
-
-A full version of the tbb project can be downloaded at
-http://threadingbuildingblocks.org/.
-
-
-./tbb41_20130401oss
-==============
+tbb41_*
+-------
```
Comment (bangerth) [inline bundled/README.md]: Just use `tbb*` without the version number.

File: bundled/README.md [hunk 5]
```diff
@@ -1,64 +1,54 @@
...
 
 Contains parts of the Intel Threading Building Blocks library copyrighted
 by the respective authors and licensed under the GNU General Public License
-(GPL) Version 2.0 with a runtime exception. See tbb41_20130401oss/COPYING
+(GPL) Version 2.0 with a runtime exception. See `tbb41_*/COPYING`
```
Comment (bangerth) [inline bundled/README.md]: Same here.

**Comment by bangerth** [comment]
Well, I do trust you guys and don't need to see every patch -- if @tjhei is fine with it, then so should I. But since you did prompt me, I had some comments anyway. Feel free to do with them whatever you want, then merge yourself.

**Pull request merged**